### PR TITLE
Adding _name_here_

### PR DESCRIPTION
**Pull request by nhibberd**
Need a name for flipr 12 fmap

### CODE REVIEW

**Comment by erikd-ambiata** [comment]
:+1: on :green_apple:

File: src/P/Functor.hs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import           Data.Function (flip)
+import           Data.Functor as X (
+                     Functor(..)
+                   , ($>)
+                   , (<$>)
+                   , void
+                   )
+
+mapf :: Functor f => f a -> (a -> b) -> f b
```
Comment (charleso) [inline src/P/Functor.hs]: Would suggestion `<&>` except that it clashes with lens :(

**Comment by damncabbage** [comment]
voting for `apfm`

**Comment by thumphries** [comment]
`ampf`

**Comment by jystic** [comment]
`ffor`

**Comment by tranma** [comment]
👍 ffor
MapReduceBigDataMapper

**Comment by jystic** [comment]
`with` would be good too, not sure if that would clash with existing names?

```hs
xs `with` \x -> x + 1
```

**Comment by EduardSergeev** [comment]
There is `<**>` for `Applicative` (flipped `<*>`), so maybe `<$$>`?

**Comment by jystic** [comment]
If we were going for an operator as well then I'd suggest `<&>` because the following functions are already in `base` and exported by `ambiata-p`:

```hs
($) :: (a -> b) -> a -> b
(&) :: a -> (a -> b) -> b

(<$>) :: Functor f => (a -> b) -> f a  -> f b
```
so naturally:
```hs
(<&>) :: Functor f => f a -> (a -> b) -> f b
```

The only problem is that lens also exports [<&>](https://hackage.haskell.org/package/lens-4.14/docs/Control-Lens-Lens.html#v:-60--38--62-) (same definition `flip fmap`) so it would break code that is currently using it.

**Comment by markhibberd** [comment]
I really don't see the an operator helping much here, not against having one as well, but would use prefix version a lot more. I am :100: for 'with',  and then maybe adding <&> if people want an operator, but not until we want to do a set of breaking change, maybe tack it onto the semigroup change and ghc-8?

**Comment by markhibberd** [comment]
(Meaning add with now, clashing operator later)

**Comment by EduardSergeev** [comment]
> I really don't see the an operator helping much here, not against having one as well, but would use prefix version a lot more

Prefix version with right-associative `($)`? Indeed `infixl` operator wouldn't be much of a help here.

If haskell supported it I would actually prefer to use unary operators to modify existing functions instead of "multiplying entities". Something you can do even now:
```
λ> :set -XPostfixOperators
λ> let (~~) = flip
λ> :t (fmap~~)
(fmap~~) :: Functor f => f a -> (a -> b) -> f b
λ> :t (mapM~~)
(mapM~~) :: (Monad m, Traversable t) => t a -> (a -> m b) -> m (t b)
λ> fmap~~ [1::Int,2,3] $ \i -> i + 1
[2,3,4]
```
But in a less savaged way.

**Pull request merged**